src/CMakeLists.txt

```diff
@@ -36,6 +36,16 @@ target_compile_definitions(${PROJECT_NAME} PRIVATE LIBM2K_EXPORTS)
 if (ENABLE_LOG)
 	target_compile_definitions(${PROJECT_NAME} PUBLIC LIBM2K_ENABLE_LOG)
 endif()
+if (MSVC)
+	add_definitions(
+			/wd4250
+			/wd4251
+			/wd4275)
+endif()
+
+if(NOT MSVC)
+	target_compile_options(${PROJECT_NAME} PRIVATE -Wall -Wextra -Wreturn-local-addr)
```

Comment:
I think this should be `-Wreturn-stack-address` not `-Wreturn-local-addr`

Reply:
-Wreturn-local-addr is a gcc flag, while -Wreturn-stack-address is a clang one, both having the same meaning.

Reply:
Well then for gcc/clang the corresponding flag should be used. One way of doing this is multiple if branches where we add specific flags for clang and for gcc having the common ones in one place (`-Wall, -Wextra`) or using generator expressions
```cmake
target_compile_options(${PROJECT_NAME}
    PRIVATE
        -Wall
        -Wextra
        $<$<LANG_COMPILER_ID:GNU>: -Wreturn-local-addr>
        $<$<LANG_COMPILER_ID:Clang, AppleClang>: -Wreturn-stack-address>
)
```
Hope i got the syntax right.

Reply:
If we are compiling with clang -Wreturn-local-addr will have no effects, and viceversa gcc with -Wreturn-stack-address

Reply:
Modified.